apollo-router/src/plugins/traffic_shaping/retry.rs

```diff
@@ -77,78 +78,12 @@ impl<E> Policy<subgraph::Request, subgraph::Response, E> for RetryPolicy {
                     .context
                     .upsert::<_, usize>(SubgraphRequestResendCountKey::new(&req.id), |val| val + 1);
 
-                Some(future::ready(self.clone()))
+                Some(future::ready(()))
             }
         }
     }
 
-    fn clone_request(&self, req: &subgraph::Request) -> Option<subgraph::Request> {
+    fn clone_request(&mut self, req: &subgraph::Request) -> Option<subgraph::Request> {
         Some(req.clone())
     }
 }
```

Comment:
why did we remove the test below?

Reply:
That feature is to be removed (soon) and so the test wasn't ported. I've ported it and added it back so the test can be removed when the feature is removed.